### ISSUE TITLE
Literal support for redundantType 

### ISSUE THREAD

**Issue by calda** [nicklockwood/SwiftFormat#920]
It would be cool if the `redundantType` rule supported literals:

### Before

```swift
let a: Bool = true
let b: String = "foo"
let c: Int = 1
let d: Double = 3.14
let e: [String] = ["bar"]
```

### After (with enhancement)

```swift
let a = true
let b = "foo"
let c = 1
let d = 3.14
let e = ["bar"]
```

It seems like this would be possible, but would probably have to be special-cased for each type

**Comment by calda** [comment]
Added in #921

**Comment by nicklockwood** [comment]
@calda I've merged this, but I'm going to defer landing it until the next major release, as some of these changes may be unexpected.

In particular, people may have added explicit types for things like array literals in order to improve compilation performance, and this will remove those.

**Comment by calda** [comment]
Should we add an option for this (maybe `--literalTypes removeRedundant` or something) to make it opt-in?

**Comment by nicklockwood** [comment]
I'm not sure. I also want to add some logic to implement the reverse transform (add explicit types for literals) and an option to always use explicit types for members and globals but use inference for locals. With those in place, there's probably no need for an opt out.

**Comment by nicklockwood** [comment]
Landed in 0.49.0
